sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1425,11 +1450,49 @@ class Analyzer(
       // rule: ResolveDeserializer.
       case plan if containsDeserializer(plan.expressions) => plan
 
-      // SPARK-25942: Resolves aggregate expressions with `AppendColumns`'s children, instead of
-      // `AppendColumns`, because `AppendColumns`'s serializer might produce conflict attribute
-      // names leading to ambiguous references exception.
-      case a @ Aggregate(groupingExprs, aggExprs, appendColumns: AppendColumns) =>
-        a.mapExpressions(resolveExpressionTopDown(_, appendColumns))
+      // SPARK-31607: Resolve Struct field in groupByExpressions and aggregateExpressions
```

Comment:
In the comment, SPARK-31607 seems to be a typo of SPARK-31670 because SPARK-31607 is `Improve the perf of CTESubstitution`.

Reply:
> In the comment, [SPARK-31607](https://issues.apache.org/jira/browse/SPARK-31607) seems to be a typo of [SPARK-31670](https://issues.apache.org/jira/browse/SPARK-31670) because [SPARK-31607](https://issues.apache.org/jira/browse/SPARK-31607) is `Improve the perf of CTESubstitution`.

Yea，sorry for my mistake

Reply:
Never mind~, @AngersZhuuuu . :)